File: app_mobile/kmp/androidApp/src/main/java/com/ansmi/gestsquadre/kmp/ui/components/TacticalComponents.kt
package com.ansmi.gestsquadre.kmp.ui.components

import androidx.compose.foundation.Image
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.BiasAlignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Shadow
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ansmi.gestsquadre.kmp.R
import com.ansmi.gestsquadre.kmp.ui.theme.TacticalFrame

private val WhiteTextShadow = TextStyle(
    shadow = Shadow(color = Color.Black, blurRadius = 8f),
)

private val WhiteTitleStyle = TextStyle(
    color = Color.White,
    fontWeight = FontWeight.ExtraBold,
    shadow = Shadow(color = Color.Black, blurRadius = 8f),
)

private val WhiteBodyStyle = TextStyle(
    color = Color.White,
    fontWeight = FontWeight.SemiBold,
    shadow = Shadow(color = Color.Black, blurRadius = 8f),
)

@Composable
fun TacticalShell(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 12.dp, vertical = 10.dp),
        contentAlignment = Alignment.TopCenter,
    ) {
        Surface(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(42.dp))
                .border(3.dp, TacticalFrame, RoundedCornerShape(42.dp)),
            shape = RoundedCornerShape(42.dp),
            color = Color.Transparent,
        ) {
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 18.dp),
            ) {
                content()
            }
        }
    }
}

@Composable
fun MainButton(
    label: String,
    backgroundColor: Color,
    foregroundColor: Color,
    onClick: (() -> Unit)?,
    modifier: Modifier = Modifier,
    fontWeight: FontWeight = FontWeight.ExtraBold,
) {
    Button(
        onClick = { onClick?.invoke() },
        enabled = onClick != null,
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        colors = ButtonDefaults.buttonColors(
            containerColor = backgroundColor,
            contentColor = foregroundColor,
            disabledContainerColor = backgroundColor,
            disabledContentColor = foregroundColor,
        ),
        contentPadding = PaddingValues(vertical = 16.dp),
    ) {
        Text(
            text = label,
            textAlign = TextAlign.Center,
            fontSize = 18.sp,
            fontWeight = fontWeight,
        )
    }
}

@Composable
fun OpenGolfLogoBanner(
    modifier: Modifier = Modifier,
    width: Float? = null,
) {
    BoxWithConstraints(modifier = modifier.fillMaxWidth()) {
        val resolvedWidth = width ?: (maxWidth.value * 0.92f).coerceAtMost(260f)
        val aspect = 840f / 200f
        val heightDp = (resolvedWidth / aspect).dp
        Surface(
            modifier = Modifier
                .align(Alignment.TopCenter)
                .width(resolvedWidth.dp)
                .shadow(10.dp, RoundedCornerShape(10.dp))
                .clip(RoundedCornerShape(10.dp)),
            color = Color.White,
            shape = RoundedCornerShape(10.dp),
        ) {
            Image(
                painter = painterResource(R.drawable.logo_open_golf_2026),
                contentDescription = "Open d'Italia 2026",
                modifier = Modifier
                    .padding(start = 5.dp, end = 11.dp, top = 5.dp, bottom = 5.dp)
                    .width((resolvedWidth - 16f).dp)
                    .height(heightDp + 4.dp),
                contentScale = ContentScale.FillWidth,
                alignment = BiasAlignment(-0.1f, 0f),
            )
        }
    }
}

@Composable
fun AppTitleBlock(modifier: Modifier = Modifier) {
    Column(
        modifier = modifier.fillMaxWidth(),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        OpenGolfLogoBanner()
        Text(
            text = "Tracking",
            textAlign = TextAlign.Center,
            style = WhiteTitleStyle.copy(fontSize = 32.sp, letterSpacing = 0.8.sp),
            modifier = Modifier.padding(top = 18.dp),
        )
        Text(
            text = "SQUADRE",
            textAlign = TextAlign.Center,
            style = WhiteTitleStyle.copy(
                fontSize = 36.sp,
                fontWeight = FontWeight.Black,
                letterSpacing = 1.4.sp,
            ),
            modifier = Modifier.padding(top = 2.dp),
        )
    }
}

@Composable
fun TacticalTitleText(
    text: String,
    modifier: Modifier = Modifier,
    fontSize: Int = 24,
) {
    Text(
        text = text,
        modifier = modifier,
        textAlign = TextAlign.Center,
        style = WhiteTitleStyle.copy(fontSize = fontSize.sp),
    )
}

@Composable
fun TacticalBodyText(
    text: String,
    modifier: Modifier = Modifier,
    fontSize: Int = 14,
    color: Color = Color.White,
) {
    Text(
        text = text,
        modifier = modifier.fillMaxWidth(),
        textAlign = TextAlign.Center,
        style = WhiteBodyStyle.copy(fontSize = fontSize.sp, color = color),
    )
}

File: app_mobile/kmp/iosApp/iosApp/TacticalComponents.swift
import SwiftUI

struct TacticalShell<Content: View>: View {
    @ViewBuilder let content: Content

    var body: some View {
        content
            .padding(.horizontal, 20)
            .padding(.vertical, 18)
            .frame(maxWidth: .infinity)
            .background(Color.clear)
            .overlay {
                RoundedRectangle(cornerRadius: 42, style: .continuous)
                    .stroke(TacticalColors.frame, lineWidth: 3)
            }
            .padding(.horizontal, 12)
            .padding(.vertical, 10)
    }
}

struct MainButton: View {
    let label: String
    let backgroundColor: Color
    let foregroundColor: Color
    let enabled: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(label)
                .font(.system(size: 18, weight: .heavy))
                .foregroundStyle(foregroundColor)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 16)
                .background(backgroundColor)
                .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        }
        .disabled(!enabled)
        .opacity(enabled ? 1 : 0.92)
    }
}

struct OpenGolfLogoBanner: View {
    var width: CGFloat?

    var body: some View {
        Group {
            if let uiImage = UIImage(named: "logo_open_golf_2026") {
                Image(uiImage: uiImage)
                    .resizable()
                    .scaledToFit()
                    .padding(.horizontal, 8)
                    .padding(.vertical, 7)
            } else {
                Text("Open d'Italia 2026")
                    .font(.headline.bold())
                    .foregroundStyle(.black)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 20)
            }
        }
        .frame(width: width)
        .frame(maxWidth: width == nil ? .infinity : nil)
        .background(Color.white)
        .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
        .shadow(color: .black.opacity(0.25), radius: 6, y: 2)
    }
}

struct AppTitleBlock: View {
    var body: some View {
        VStack(spacing: 0) {
            OpenGolfLogoBanner()
            Text("Tracking")
                .font(.system(size: 32, weight: .heavy))
                .foregroundStyle(.white)
                .shadow(color: .black, radius: 4, y: 1)
                .padding(.top, 18)
            Text("SQUADRE")
                .font(.system(size: 36, weight: .black))
                .foregroundStyle(.white)
                .shadow(color: .black, radius: 4, y: 1)
                .padding(.top, 2)
        }
        .frame(maxWidth: .infinity)
    }
}

struct TacticalTitleText: View {
    let text: String

    var body: some View {
        Text(text)
            .font(.system(size: 24, weight: .heavy))
            .foregroundStyle(.white)
            .shadow(color: .black, radius: 4, y: 1)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
    }
}

struct TacticalBodyText: View {
    let text: String
    var fontSize: CGFloat = 14
    var color: Color = .white

    var body: some View {
        Text(text)
            .font(.system(size: fontSize, weight: .semibold))
            .foregroundStyle(color)
            .shadow(color: .black.opacity(0.8), radius: 3, y: 1)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
    }
}

struct TocNotificationPanel: View {
    let message: String?

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: 10, style: .continuous)
                .fill(TacticalColors.navy)
                .overlay {
                    RoundedRectangle(cornerRadius: 10, style: .continuous)
                        .stroke(Color.white.opacity(0.28), lineWidth: 1)
                }
            if let message, !message.isEmpty {
                Text(message)
                    .font(.system(size: 14, weight: .bold))
                    .foregroundStyle(.white)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 14)
                    .padding(.vertical, 12)
            }
        }
        .frame(minHeight: 72)
        .frame(maxWidth: .infinity)
    }
}

struct SquadBlockingAlert: View {
    let message: String

    var body: some View {
        Text(message)
            .font(.system(size: 16, weight: .bold))
            .foregroundStyle(.white)
            .multilineTextAlignment(.center)
            .frame(maxWidth: .infinity)
            .padding(.horizontal, 14)
            .padding(.vertical, 12)
            .background(TacticalColors.red)
            .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
    }
}
